rpc/json-rpc/src/error/code.rs

```diff
@@ -0,0 +1,224 @@
+//! Error codes.
+
+//---------------------------------------------------------------------------------------------------- Use
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::error::constants::{
+    INTERNAL_ERROR, INVALID_PARAMS, INVALID_REQUEST, METHOD_NOT_FOUND, PARSE_ERROR, SERVER_ERROR,
+};
+
+//---------------------------------------------------------------------------------------------------- ErrorCode
+/// [Error object code](https://www.jsonrpc.org/specification#error_object).
+///
+/// This `enum` encapsulates JSON-RPC 2.0's error codes
+/// found in [`ErrorObject`](crate::error::ErrorObject).
+///
+/// It associates the code integer ([`i32`]) with its defined message.
+///
+/// # Application defined errors
+/// The custom error codes past `-32099` (`-31000, -31001`, ...)
+/// defined in JSON-RPC 2.0 are not supported by this enum because:
+///
+/// 1. The `(i32, &'static str)` required makes the enum more than 3x larger
+/// 2. It is not used by Cuprate/Monero
```

Comment:
I don't think we'll use custom errors, right?

Reply:
If monerod doesn't then probably not